js/web/lib/wasm/session-handler.ts

```diff
@@ -110,19 +110,31 @@ export class OnnxruntimeWebAssemblySessionHandler implements SessionHandler {
   outputNames: string[];
   private outputNamesUTF8Encoded: number[];
 
-  loadModel(model: Uint8Array): void {
+  loadModel(model: Uint8Array, options?: InferenceSession.SessionOptions): void {
     const wasm = getInstance();
     if (!ortInit) {
-      wasm._OrtInit(env.wasm.numThreads!, 2 /* LoggingLevel::Warning */);
+      const errorCode = wasm._OrtInit(env.wasm.numThreads!, this.getLoggingLevel(env.wasm.loggingLevel!));
+      if (errorCode !== 0) {
+        throw new Error(`Can't initialize onnxruntime. error code = ${errorCode}`);
+      }
       ortInit = true;
     }
 
     const modelDataOffset = wasm._malloc(model.byteLength);
+
+    const allocs: number[] = [];
+    const sessionOptionsHandle = this.setSessionOptions(allocs, options);
+    if (sessionOptionsHandle === 0) {
+      throw new Error('Can\'t create session options');
+    }
+
     try {
       wasm.HEAPU8.set(model, modelDataOffset);
-      this.sessionHandle = wasm._OrtCreateSession(modelDataOffset, model.byteLength);
+      this.sessionHandle = wasm._OrtCreateSession(modelDataOffset, model.byteLength, sessionOptionsHandle);
     } finally {
       wasm._free(modelDataOffset);
+      wasm._OrtReleaseSessionOptions(sessionOptionsHandle);
+      allocs.forEach(i => wasm._free(i));
```

Comment:
not sure if we can do `allocs.forEach(wasm._free)`

Reply:
we can do like that